hw/ip/spi_device/doc/programmers_guide.md

```diff
@@ -40,6 +40,20 @@ The SRAM begins at `0x1000`, which in the figure is `0x000`.
 The regions starting from `0xF00` to `0xFFF` are assigned to TPM Read/Write FIFOs.
 They are not used in this version of IP.
 
+## SPI Mode Changes
+
+The clock for port B of the dual-port SRAM comes from different sources, depending on which SPI mode is selected.
+For the generic mode, the peripheral clock is used, while for flash and passthrough mode, the SPI clock is used.
+Since the SPI clock is not active when the SPI interface is idle, the design cannot make use of a glitch-free clock mux that would require a continuously running clock.
+Therefore, software has to use the [`CONTROL.sram_clk_en`](../data/spi_device.hjson#control) register to gate the clock while switching between modes with different dual-port SRAM clock sources.
+
+For any mode change that involves switching to a different clock source, the following programming sequence must be followed:
+
+1. Check if SPI line is idle.
```

Comment:
nit: This doesn't really make sense. Software must be able to ensure that no transactions can be incoming. Otherwise, there would be race conditions that negate the value of this.

Reply:
Oops, I mean that either software needs to hold the upstream SPI host in reset or somehow signal it to hold off. There must be a gate that doesn't rely on a point-in-time check of chip-select, which could be incorrect by the next clock cycle.

Reply:
Yeah that's right - this needs to be done at the protocol level.

I see this as a known limitation for the shuttle that we can work around in controlled settings.

For prod, this limitation should be removed (e.g. by changing the DPSRAM and async FIFO architecture underneath).

Reply:
What I meant is that this guidance is misleading, and we should change this line. Software checking "if the SPI line is idle" is not sufficient to safely make the clock switch.

Reply:
Ack, I'll also add a note that this limitation is likely going to be removed in the future.